nixos/modules/virtualisation/xen-dom0.nix

```diff
@@ -267,26 +267,36 @@ in
         mkdir -p /var/lib/xen # so we create them here unconditionally.
         grep -q control_d /proc/xen/capabilities
         '';
-      serviceConfig.ExecStart = ''
-        ${cfg.stored}${optionalString cfg.trace " -T /var/log/xen/xenstored-trace.log"} --no-fork
-        '';
+      serviceConfig = if cfg.package.version < "4.8" then
+	{ ExecStart = ''
+	    ${cfg.stored}${optionalString cfg.trace " -T /var/log/xen/xenstored-trace.log"} --no-fork
```

Comment:
The code that you are referring to has been carried over from the service that was created for xen 4.5 (or earlier). An argument for keeping it is to try to minimise the change. If you insist, then I can remove it, but otherwise we may keep it as is and drop it when we drop xen 4.5.